Unable to compile ESP32 examples with CMake with latest CHIP master

#### Problem
Unable to compile all-clusters-app, temperature-measurement-app and lock-app for ESP32 using CMake. The same work fine using automake. 
This seems to be a regression after the recent changes in the interaction model

Compilation error seen:
```
esp-idf/chip/lib/libCHIP.a(libCHIPDataModel.CommandHandler.cpp.o):(.literal._ZN4chip3app14CommandHandler25ProcessCommandDataElementERNS0_18CommandDataElement6ParserE+0x8): undefined reference to `chip::app::DispatchSingleClusterCommand(unsigned short, unsigned char, unsigned char, chip::TLV::TLVReader&, chip::app::Command*)'
```

--- a/config/esp32/components/chip/CMakeLists.txt
+++ b/config/esp32/components/chip/CMakeLists.txt
@@ -178,6 +178,9 @@ if(NOT CONFIG_USE_MINIMAL_MDNS)
     list(APPEND chip_libraries $<TARGET_FILE:${mdns_lib}>)
 endif()
 
+idf_component_get_property(main_lib main COMPONENT_LIB)
+list(APPEND chip_libraries $<TARGET_FILE:${main_lib}>)
+
 target_link_libraries(${COMPONENT_LIB} INTERFACE -Wl,--start-group 
                                                 ${chip_libraries}
                                                 $<TARGET_FILE:mbedcrypto> $<TARGET_FILE:${esp32_mbedtls_lib}> 

